src/main/java/seedu/us/among/ui/EndpointListPanel.java

```diff
@@ -41,7 +42,7 @@ protected void updateItem(Endpoint endpoint, boolean empty) {
                 setGraphic(null);
                 setText(null);
             } else {
-                setGraphic(new EndpointCard(endpoint, getIndex() + 1).getRoot());
+                Platform.runLater(() -> setGraphic(new EndpointCard(endpoint, getIndex() + 1).getRoot()));
```

Comment:
The reason for updating the endpoint card later is because?

Reply:
This was required to address the error raised: `not on FX application thread; currentThread = Thread-4`